nswbuff.el

```diff
@@ -632,10 +638,13 @@ BUFFER should be a buffer name.  It is tested against the regular expressions in
 This function is called directly by the nswbuff timer."
   (let ((buffer (get-buffer (nswbuff-status-buffer-name)))
         (buffer-list (nreverse nswbuff-initial-buffer-list)))
-    (if (and (window-live-p nswbuff-status-window)
-             (not (eq nswbuff-status-window-layout 'minibuffer)))
+    ;; Cleanup status window and status buffer
+    (if (eq nswbuff-status-window-layout 'minibuffer)
+        (with-current-buffer buffer (erase-buffer))
+      (when (window-live-p nswbuff-status-window)
         (delete-window nswbuff-status-window))
-    (if buffer (with-current-buffer buffer (erase-buffer)))
+      (when buffer
+        (kill-buffer buffer)))
```

Comment:
Couldn't we keep using `erase-buffer`? Buffer switching happens a gazillion times a day, it doesn't seem wrong to keep the buffer around.

Reply:
Yes it should be doable. I kept the `kill-buffer` to be sure to introduce no logic change with before. 
For instance this `kill-buffer` is expected by `nswbuff-get-status-buffer` because it does a `(if (buffer-live-p nswbuff-status-buffer) ...`

So if this logic is changed, then more code has to be changed, which can be tricky as we have just seen with this minibuffer cleaning issue. So it's up to you, I can give it a try if you want.

Reply:
> So it's up to you, I can give it a try if you want.

Only if you're itching to do it. :slightly_smiling_face: `nswbuff` and `swbuff` before it have been doing fine with `kill-buffer` for over two decades, so I don't think there's really a practical advantage to it. It just seems conceptually better.

Reply:
Let me know if you do decide to change it, then I'll hold off on merging this PR.

Reply:
Actually, I won't hold off, because it fixes a bug.

Reply:
Yes I think using the `kill-buffer` is fine too. 

When I have more time I may give it a try, I think I would do a bit of refactoring at the same time if I really look into it.

Anyway thank you for reviewing quickly the changes of this minibuffer layout features !